clients/src/main/java/org/apache/kafka/clients/admin/Admin.java

```diff
@@ -233,41 +232,41 @@ default DeleteTopicsResult deleteTopics(Collection<String> topics) {
      * @return The DeleteTopicsResult.
      */
     DeleteTopicsResult deleteTopics(Collection<String> topics, DeleteTopicsOptions options);
-    
+
     /**
-     * This is a convenience method for {@link #deleteTopicsWithIds(Collection, DeleteTopicsOptions)}
+     * This is a convenience method for {@link #deleteTopics(TopicCollection, DeleteTopicsOptions)}
      * with default options. See the overload for more details.
      * <p>
-     * This operation is supported by brokers with version 2.8.0 or higher.
+     * This operation is supported by brokers with version 3.0.0 or higher.
      *
-     * @param topics The topic IDs for the topics to delete.
-     * @return The DeleteTopicsWithIdsResult.
+     * @param topics The topics to delete.
+     * @return The DeleteTopicsResult.
      */
-    default DeleteTopicsWithIdsResult deleteTopicsWithIds(Collection<Uuid> topics) {
-        return deleteTopicsWithIds(topics, new DeleteTopicsOptions());
+    default DeleteTopicsResult deleteTopics(TopicCollection topics) {
+        return deleteTopics(topics, new DeleteTopicsOptions());
     }
 
     /**
      * Delete a batch of topics.
      * <p>
      * This operation is not transactional so it may succeed for some topics while fail for others.
      * <p>
-     * It may take several seconds after the {@link DeleteTopicsWithIdsResult} returns
+     * It may take several seconds after the {@link DeleteTopicsResult} returns
      * success for all the brokers to become aware that the topics are gone.
      * During this time, {@link #listTopics()} and {@link #describeTopics(Collection)}
      * may continue to return information about the deleted topics.
      * <p>
-     * If delete.topic.enable is false on the brokers, deleteTopicsWithIds will mark
+     * If delete.topic.enable is false on the brokers, deleteTopics will mark
      * the topics for deletion, but not actually delete them. The futures will
      * return successfully in this case.
      * <p>
-     * This operation is supported by brokers with version 2.8.0 or higher.
+     * This operation is supported by brokers with version 3.0.0 or higher.
      *
-     * @param topics  The topic IDs for the topics to delete.
+     * @param topics  The topics to delete.
      * @param options The options to use when deleting the topics.
-     * @return The DeleteTopicsWithIdsResult.
+     * @return The DeleteTopicsResult.
      */
-    DeleteTopicsWithIdsResult deleteTopicsWithIds(Collection<Uuid> topics, DeleteTopicsOptions options);
+    DeleteTopicsResult deleteTopics(TopicCollection topics, DeleteTopicsOptions options);
```

Comment:
Since `TopicCollection` can handle both "by name" and "by ID" deletion, is the intent to eventually deprecate the old deletion api or will we keep it around for convenience?

Reply:
Eventually we want to deprecate the old api yes